### ISSUE TITLE
No upgrades to elementum and burst after kodi's upgrade to v19

### ISSUE THREAD

**Issue by pitsi** [elgatito/plugin.video.elementum#724]
Preface. I upgraded to kodi 19 a few weeks ago (right after the day it was released). After the upgrade was completed (the dialogs about updating databases before launch and the mandatory upgrades of all the addons), I checked if every third party addon and repo is still there, and indeed everything was there and the repos seemed to work just fine, e.g. I could see elementum and burst listed under elementum's repo. 

## Expected Behavior
So, since the repo is still enabled, one would expect the updates to keep coming once they are available.

## Current Behavior
However, this does not happen and I assume it's because of some new "policy" about third party repos that the new kodi implements.

## Possible Solution
As seen in the screenshots below, reinstalling each addon again upgrades them. I have to wait a few more days to see if they will get autoupgraded

![screenshot00000](https://user-images.githubusercontent.com/5880037/111951088-d5921380-8aeb-11eb-92a7-53482768d1c8.png)

![screenshot00001](https://user-images.githubusercontent.com/5880037/111951098-da56c780-8aeb-11eb-838c-2ebc39c1f08b.png)

## Steps to Reproduce (for bugs)
There is a small description of what I did in the preface. Do not do it unless you really want to go through installing 18.9 fresh, then install elementum on it and finally upgrade to 19, because the procedure is quite tedious.

## Context
I saw earlier in reddit that elementum was upgraded to 0.1.73 and burst to 0.63, so I checked where my kodi installations are.
I also have another kodi 18.9 installation that works as it should and it is now on elementum 0.1.72 and burst 0.63.
One more annoying thing after the manual upgrage was that kodi downloaded the zip that containes elementum for all operating systems and architectures. ~140MB as seen below, while I was expecting for something around 15MB
```
$ ls -l --block-size=k .kodi/addons/packages/
total 143480K
-rw-r--r-- 1 x x 139949K Mar 22 08:31 plugin.video.elementum-0.1.72.zip
-rw-r--r-- 1 x x   3523K Mar 22 08:32 script.elementum.burst-0.0.63.zip
```
On top of that, ALL the binaries were extracted in ~/.kodi/addons/plugin.video.elementum/resources/bin, raising that folder's size to 350+MB
```
ncdu 1.15.1 ~ Use the arrow keys to navigate, press ? for help                  
--- /home/x/.kodi/addons/plugin.video.elementum/resources/bin ----------------
                         /..                                                    
   45.8 MiB [##########] /android_x64
   43.1 MiB [######### ] /android_arm64
   37.6 MiB [########  ] /android_x86
   36.9 MiB [########  ] /darwin_x64
   35.4 MiB [#######   ] /android_arm
   28.3 MiB [######    ] /windows_x64
   27.8 MiB [######    ] /linux_x64
   25.4 MiB [#####     ] /windows_x86
   25.3 MiB [#####     ] /linux_arm64
   25.3 MiB [#####     ] /linux_x86
   22.8 MiB [####      ] /linux_armv6
   21.7 MiB [####      ] /linux_armv7
    4.0 KiB [          ] /client
```
I will delete the ones that are not linux_x64, but I assumed it would download and install only the one for the os and the architecture it is currently run on.

## Your Environment
* Version used: 0.1.69 upgraded to 0.1.72
* Environment name and version: Kodi 19 on debian testing/unstable x64.
* Operating System and version: Debian testing/unstable x64
* Link to Kodi debug log: Not needed.

**Comment by elgatito** [comment]
> However, this does not happen and I assume it's because of some new "policy" about third party repos that the new kodi implements.

I don't know about any changes in repositories with Kodi 19.

> I saw earlier in reddit that elementum was upgraded to 0.1.73 and burst to 0.63

No, it does not. https://github.com/elgatito/plugin.video.elementum/releases

> I will delete the ones that are not linux_x64, but I assumed it would download and install only the one for the os and the architecture it is currently run on.

No, it always download archive will all architectures, there were reasons for that.

**Comment by pitsi** [comment]
> I don't know about any changes in repositories with Kodi 19.
Me neither. I will look the rest of my addons and report back. I have... 1 more third party repo on my pc right now. 
Elementum and burst are now shown as installed in the repo's listing. Before reinstalling them, there was no notification next to each one, not even "update available". If there really is a new "policy", I think it should be mentioned in the readme, just for the ones that will upgrade from 18.9 to 19.

![screenshot00000](https://user-images.githubusercontent.com/5880037/111955922-cf536580-8af2-11eb-8d07-35b1b978da50.png)
![screenshot00001](https://user-images.githubusercontent.com/5880037/111955929-d37f8300-8af2-11eb-8abb-03ff048a45fe.png)

As for the rest, reddit was probably wrong and thank you for the clarification.

**Comment by antonsoroko** [comment]
@pitsi 
could you please share a screen with addons-related settings in kodi? (system->addons)

side note: instead of checking reddit for new releases you can subscribe for notifications about new releases here in github - see "watch" button in the right top corner of the page (you can choose only releases).

**Comment by pitsi** [comment]
There you go.
![screenshot00002](https://user-images.githubusercontent.com/5880037/111962338-de3e1600-8afa-11eb-803c-aac18f404f88.png)

As for the updates, I use this to get updates via rss in my thunderbird
`https://github.com/elgatito/plugin.video.elementum/releases.atom`
along with 15+ more github projects that I am interested in.
But it is almost pointless to have, because I open my other kodi installation (libreelec 9.2.6 x64, kodi 18.9) like 10 times a day, and I am sure It checks for updates at least half of those times and on startup.

**Comment by pitsi** [comment]
So, I have one addon from that other third party repo. It was last updated ~10 days before the release date of v19, but since it is python3 ready, it showed up as installed and it works as it should. 
However, like elementum, it does not show "enabled" next to its name, it's just blank. I will not reinstall it and I will be keeping an eye on what kodi will do once an update for it becomes available.

On the other hand, I want to poke things a little and see how will kodi react. On a scale from 1 to 10, how much do you think I will f up my installation if I delete ~/.kodi/userdata/Database/Addons33.db and force kodi make it again?
This is the file that acts like a registry to addons, e.g. burst was downloaded from here, it is made by this dev, it has this icon etc.

**Comment by elgatito** [comment]
@pitsi you screenshot shows "Official repositories only". I believe it means Kodi official repo, not addon's.

**Comment by antonsoroko** [comment]
Update official add-ons from
Official repositories only: add-ons from official repositories will be prevented from being auto-updated from private repositories (default).
https://kodi.wiki/view/Add-on_manager

So this should not be a problem, I think.


About deletion - people says it will disable all add-ons, e.g. https://discourse.osmc.tv/t/need-to-delete-addons27-db-and-all-its-gone-for-ever/75698/5

**Comment by pitsi** [comment]
The other option there is "any repositories" and the explanation below says

> By default, **add-ons from official repositories will be prevented from being auto-updated from private repositories**. For cases such as updating from an add-ons beta repository this option can be switched to [Any repositories] (bear in mind this is a less secure option and enabling it could cause incompatibility and crashes).

Notice that thing in bold? I think it only refers to official addons, e.g. youtube, twitch etc and since elementum is not an official one, it does not fall in that category.

Brb, I am off to mess things up! Thank you for the clarifications.

**Comment by pitsi** [comment]
Done! Kodi started with estuary and asked me to enable every single addon on the next launch and it even showed the options for the ones that were not set to defaults (elementum, burst, youtube, twitch etc). I enabled them all and switched the forementioned setting to "any repositories" just in case. And things got back to what I reported when I first opened this issue.
ALL my addons, even the official ones, now have nothing next to their name and I will have to reinstall them if I want to get updates. Actually, only the addons that refer to repositories have "enabled" next to their name.
I am always talking about the ones listed addons > install from repository > some repository > some category > addon name.
The list under addons > my addons > all or any other subset > addon name has "enabled" next to every single installed addon, official or third party.

On the scale from 1 to 10 above, I think this is a 8.5 :D
Feel free to close this issue if you also consider this is not an elementum issue but a kodi one.

**Comment by antonsoroko** [comment]
Looks like the the migration from old kodi to new has some issues.
I guess you can report this to kodi developers.

**Comment by pitsi** [comment]
You do know that they will go nuts when they hear about third party addons, right?
Back in 17, I had reported that when deleting addons.db, the file gets recreated but the third party addons stay disabled (an issue that did not happen until 16) and they just said that this was intentional. So, judging of what 19 does by asking what addons do I want to be enabled, they will probably say I should be thankful for that change.

**Comment by pitsi** [comment]
So, I got an update today, the very first one after deleting addons.db and starting fresh. Sadly, it was for an addon that is in the main kodi repo and I am sure it would be done anyway.
I also installed another third party repo that provides a couple of subtitle addons and I expected it to auto update the one I had already installed (via zip). It did not, so I am certain all the forementioned stuff is actually a kodi issue instead of one in elementum.

Because of that, I am now really intrigued in updating libreelec (the x64 one, not the one on my rpi3b+) to v10 that has kodi 19, although it is still in beta and its devs advise for a fresh install. The amount of addons I have installed there is like 3 times the ones I have on my pc, so it will be a lot of work if something breaks and I have to start fresh. On the other hand, I already know which ones are ready for 19 and which ones are not (and the ones that will never be), so that makes my work a bit easier. I will install the repos first, buy extracting everything in the right folder and then each and every addon from its repo.

Tldr of the above. I know that elementum's repo is installed from the elemenum itself (as a video plugin). Is there a repository.elementum.zip that will save me some work?

**Comment by antonsoroko** [comment]
if i understand correctly:
elementum repo is virtual.
is refers to localhost where elementum daemon dynamically serves the data. it mirrors data from github releases.
so even if you zip your current `.kodi/addons/repository.elementum/` it will not work without already running elementum.

**Comment by pitsi** [comment]
@elgatito 
I have changed that setting to "any repositories", I mentioned it above.

@antonsoroko 
I understood that too...

So, right now I am trying something insane! I will remove all my addons from ~/.kodi/addons, except from the repository ones, by deleting their folders from there and the addons.db. I am also keeping their settings intact in ~/.kodi/addons/userdata/addon_data.
Once I am done, I will try installing them one by one from their repos.

**Comment by elgatito** [comment]
@pitsi I don't understand what you want to achieve with all those actions?

**Comment by pitsi** [comment]
I want to make a "fresh" install (not actually fresh, but an existing one with all my addons installed fresh) in which every addon is listed as installed and will get updates in the future, because kodi does not mark them as installed. And I almost did it! 
I deleted all my addons and just left all their repositories there. Then I launched kodi and it asked me "do you want to enable this repository" for every one of them. Once it was done, I let it do its updates and scan all of them and I restarted it.

After that, I batch installed my addons from command line (howto upon request), leaving elementum for the end due to its uniqueness. I downloaded the right zip as usual, installed it via "install from zip", restarted kodi and the notification "Elementum daemon has started" made me happy for like 30 seconds. Elementum is now properly installed and running, as you can see here.

![screenshot00000](https://user-images.githubusercontent.com/5880037/112332094-20fe1a80-8cc2-11eb-822e-ca71b5a398b4.png)

Then the prompt to install burst came up, I pressed yes and nothing happend. So I went to settings > addons > install from repository > elementum repository > program addons to install it manually. And this came up, although both elementum and the repo were installed already!

![screenshot00001](https://user-images.githubusercontent.com/5880037/112332524-7a664980-8cc2-11eb-959f-e50c59e328e8.png)

Obviously, I will press ok there and it will install as usual, although it will get me back to what I described in the first place.

**Comment by pitsi** [comment]
Upgrading libreelec to 10 failed miserably and now the system starts with a black screen with a cursor in the middle.
I wont't avod the fresh installation as it seems. So much wasted time :(

**Comment by pitsi** [comment]
So, a friend of mine just reported a similar behavior.
A third party addon he has installed, has not been updated since he upgraded kodi 18 to 19. It's repo is there, it is enabled, but the addon shows up as not installed and thus it is not updated.

I am now certain this is a kodi issue and although I do want them to fix it, I won't report it because of their attitude. If anyone else wants to do it, I can provide any needed support (except full logs). Feel free to close this issue anytime you wish.

**Comment by pitsi** [comment]
Time to close this, because it is not an issue of elementum but a general behavior of kodi against third party addons.
My forementioned offer stands though.
